velox/expression/tests/ArrayProxyTest.cpp

```diff
@@ -464,5 +466,129 @@ TEST_F(ArrayProxyTest, nestedArrayE2E) {
   assertEqualVectors(result, makeNestedArrayVector<int64_t>(expected));
 }
 
+TEST_F(ArrayProxyTest, copyFromEmptyArray) {
+  auto [result, writer] = makeTestWriter();
+
+  auto& proxy = writer->current();
+  std::vector<int64_t> data = {};
+
+  proxy.copy_from(data);
+  writer->commit();
+  writer->finish();
+
+  assertEqualVectors(result, makeNullableArrayVector<int64_t>({{}}));
+}
+
+TEST_F(ArrayProxyTest, copyFromIntArray) {
+  auto [result, writer] = makeTestWriter();
+
+  auto& proxy = writer->current();
+  std::vector<int64_t> data = {1, 2, 3, 4};
+
+  proxy.copy_from(data);
+  writer->commit();
+  writer->finish();
+
+  assertEqualVectors(result, makeNullableArrayVector<int64_t>({{1, 2, 3, 4}}));
+}
+
+TEST_F(ArrayProxyTest, copyFromStringArray) {
+  auto result =
+      prepareResult(std::make_shared<ArrayType>(ArrayType(VARCHAR())));
+
+  exec::VectorWriter<ArrayProxyT<Varchar>> writer;
+  writer.init(*result->as<ArrayVector>());
+  writer.setOffset(0);
+
+  auto& arrayProxy = writer.current();
+  std::vector<std::string> data = {"hi", "welcome"};
+  arrayProxy.copy_from(data);
+
+  writer.commit();
+  writer.finish();
+  auto expected = std::vector<std::vector<std::optional<StringView>>>{
+      {"hi"_sv, "welcome"_sv}};
+  assertEqualVectors(result, makeNullableArrayVector(expected));
+}
+
+TEST_F(ArrayProxyTest, copyFromNestedArray) {
+  auto elementType =
+      ArrayType(std::make_shared<ArrayType>(ArrayType(BIGINT())));
+  auto result = prepareResult(std::make_shared<ArrayType>(elementType));
+
+  exec::VectorWriter<ArrayProxyT<ArrayProxyT<int64_t>>> writer;
+  writer.init(*result.get()->as<ArrayVector>());
+  writer.setOffset(0);
+
+  std::vector<std::vector<int64_t>> data = {{}, {1, 2, 3, 4}, {1}};
+  auto& arrayProxy = writer.current();
+  arrayProxy.copy_from(data);
+
+  writer.commit();
+  writer.finish();
+
+  using array_type = std::optional<std::vector<std::optional<int64_t>>>;
+  array_type array1 = {{}};
+  array_type array2 = {{1, 2, 3, 4}};
+  array_type array3 = {{1}};
+
+  assertEqualVectors(
+      result, makeNestedArrayVector<int64_t>({{array1, array2, array3}}));
+}
+
+auto makeCopyFromTestData() {
+  std::vector<folly::F14FastMap<int64_t, int64_t>> data;
```

Comment:
We've talked in team meetings about removing dependencies on folly in Velox.  I know the intended use case for this will use F14FastMap, so we want to make sure this works for them, but for the purposes of testing, could we use an std map here?

Reply:
okiz